error[E0277]: expected a `Fn<()>` closure, found `T`
  --> $DIR/issue-68645-codegen-fulfillment.rs:14:18
   |
LL |     type F<'a> = Self;
   |                  ^^^^ expected an `Fn<()>` closure, found `T`
   |
   = note: wrap the `T` in a closure with no arguments: `|| { /* code */ }`
note: required by a bound in `Fun::F`
  --> $DIR/issue-68645-codegen-fulfillment.rs:6:17
   |
LL |     type F<'a>: Fn() -> u32;
   |                 ^^^^^^^^^^^ required by this bound in `Fun::F`
help: consider restricting type parameter `T`
   |
LL | impl<T: std::ops::Fn<()>> Fun for T {
   |       ++++++++++++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
